ludwig/utils/misc_utils.py

```diff
@@ -33,6 +34,11 @@ def set_random_seed(random_seed):
     os.environ["PYTHONHASHSEED"] = str(random_seed)
     random.seed(random_seed)
     numpy.random.seed(random_seed)
+    torch.manual_seed(random_seed)
+    if torch.cuda.is_available():
+        torch.cuda.manual_seed(random_seed)
+        torch.backends.cudnn.deterministic = True
```

Comment:
We should be careful about setting cudnn to deterministic, as that will restrict parallelism (most non-determinism in CUDA comes from mutli-threading). My opinion is that we should move this line (and the benchmark line just below) to the `initialize_pytorch` function in `torch_utils.py`, and only set this when `allow_parallel_threads` is `False`. This is consistent with the desired effect of setting that param, which is to restrict parallelism to enforce determinism.

https://github.com/ludwig-ai/ludwig/blob/master/ludwig/utils/torch_utils.py#L236

Reply:
@tgaddair   I understand you want me to move the setting of the `torch.backend.cudnn....` options to be in the section of code in `torch_utils.py` that is controlled by the `allow_parallel_threads`.  

Is this correct understanding?

Reply:
Hey @jimthompson5802, yes that would be my preference. It might make sense at some point to change `allow_parallel_threads` to `deterministic` or similar (though truly guaranteeing determinism, particularly with gpus or distributed systems, is extremely difficult). But for now I believe this is the most correct place to ensure we aren't incurring a performance penalty.

Reply:
@tgaddair   Got will make the change later tonight.  Thank you for the clarification.

Reply:
incorporated change.